polkadot/zombienet-sdk-tests/tests/functional/approval_voting_coalescing.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Test that checks approval voting coalescing does not lag finality.
+
+use anyhow::anyhow;
+
+use crate::helpers::{
+	assert_finality_lag_less_than, assert_finalized_block_height, assert_para_throughput,
+};
+use polkadot_primitives::Id as ParaId;
+use serde_json::json;
+use subxt::{OnlineClient, PolkadotConfig};
+use zombienet_sdk::NetworkConfigBuilder;
+
+#[tokio::test(flavor = "multi_thread")]
+async fn approval_voting_coalescing_test() -> Result<(), anyhow::Error> {
+	let _ = env_logger::try_init_from_env(
+		env_logger::Env::default().filter_or(env_logger::DEFAULT_FILTER_ENV, "info"),
+	);
+
+	let images = zombienet_sdk::environment::get_images_from_env();
+	let no_show_slots = 4;
+	let config = NetworkConfigBuilder::new()
+		.with_relaychain(|r| {
+			let r = r
+				.with_chain("rococo-local")
+				.with_default_command("polkadot")
+				.with_default_image(images.polkadot.as_str())
+				.with_default_args(vec![("-lparachain=debug,runtime=debug").into()])
+				.with_genesis_overrides(json!({
+					"configuration": {
+						"config": {
+							"needed_approvals": 4,
+							"relay_vrf_modulo_samples": 6,
+							"no_show_slots": no_show_slots,
+							"approval_voting_params": {
+								"max_approval_coalesce_count": 5
+							}
+						}
+					}
+				}))
+				.with_node(|node| node.with_name("validator-0"));
+
+			(1..12)
+				.fold(r, |acc, i| acc.with_node(|node| node.with_name(&format!("validator-{i}"))))
+		})
+		.with_parachain(|p| {
+			p.with_id(2000)
+				.with_default_command("undying-collator")
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.cumulus_based(false)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-undying-2000"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2001)
+				.with_default_command("undying-collator")
+				.with_default_image(
+					std::env::var("COL_IMAGE")
+						.unwrap_or("docker.io/paritypr/colander:latest".to_string())
+						.as_str(),
+				)
+				.cumulus_based(false)
+				.with_default_args(vec![("-lparachain=debug").into()])
+				.with_collator(|n| n.with_name("collator-undying-2001"))
+		})
+		.with_parachain(|p| {
+			p.with_id(2002)
+				.with_default_command("undying-collator")
+				.with_default_image(
```

Comment:
Can't we dedup ? Feels weird to just copy paste all of these para config.

Reply:
I kind of copy-pasted from other tests @pepoviola is there a more elegant way to express all this identical parachains where just the id differs ?

Reply:
:D never mind, I was lazy it is easy to avoid duplication :D